### PR TITLE
Refactor plugin manager

### PR DESCRIPTION
**Pull request by kaklakariada**
This simplifies the Plugin interface by returning an empty `Optional` instead of `null` or throwing an exception.

This requires adapting third party plugins and rebuilding plugins installed to `~/.whiterabbit/plugins/`.

### CODE REVIEW

File: logic/src/test/java/org/itsallcode/whiterabbit/logic/service/plugin/PluginManagerTest.java [hunk 1]
```diff
@@ -101,19 +113,23 @@ private void simulatePlugins(PluginWrapper... plugins)
     void getProjectReportExporter_noPluginAvailable_throwsException()
     {
         simulatePlugins();
-        assertThatThrownBy(() -> pluginManager.getProjectReportExporter("unknown"))
-                .isInstanceOf(IllegalStateException.class)
-                .hasMessage("Plugin 'unknown' not found");
+        assertThat(pluginManager.getAllFeatures(ProjectReportExporter.class)).isEmpty();
+        // assertThatThrownBy(() -> pluginManager.get("unknown"))
+        // .isInstanceOf(IllegalStateException.class)
+        // .hasMessage("Plugin 'unknown' not found");
```
Comment (ckunki) [inline logic/src/test/java/org/itsallcode/whiterabbit/logic/service/plugin/PluginManagerTest.java:119]: dead code

File: logic/src/test/java/org/itsallcode/whiterabbit/logic/service/plugin/PluginManagerTest.java [hunk 2]
```diff
@@ -101,19 +113,23 @@ private void simulatePlugins(PluginWrapper... plugins)
...
         when(plugin1.supports(ProjectReportExporter.class)).thenReturn(false);
-        assertThatThrownBy(() -> pluginManager.getProjectReportExporter("plugin1"))
-                .isInstanceOf(IllegalStateException.class)
-                .hasMessage("Plugin 'plugin1' does not support feature " + ProjectReportExporter.class.getName());
+        assertThat(pluginManager.getAllFeatures(ProjectReportExporter.class)).isEmpty();
+        // assertThatThrownBy(() ->
+        // pluginManager.getProjectReportExporter("plugin1"))
+        // .isInstanceOf(IllegalStateException.class)
+        // .hasMessage("Plugin 'plugin1' does not support feature " +
+        // ProjectReportExporter.class.getName());
```
Comment (ckunki) [inline logic/src/test/java/org/itsallcode/whiterabbit/logic/service/plugin/PluginManagerTest.java:132]: dead code

**Review by ckunki — CHANGES_REQUESTED** [review]
Wow - big change.
Looks good to me, except some dead code portions.

File: logic/src/test/java/org/itsallcode/whiterabbit/logic/service/plugin/PluginManagerTest.java (hunk 1 above)
Comment (kaklakariada) [reply logic/src/test/java/org/itsallcode/whiterabbit/logic/service/plugin/PluginManagerTest.java:119]: Thank you for seeing this :)

**Pull request merged**